enumerable_method.rb

```diff
@@ -0,0 +1,117 @@
+# Enumerable methods
+module Enumerable
+  def my_each
+    p enum_for(:my_each) unless block_given?
+    for value in self do
+      yield value
+    end
+    p self
+  end
+
+  def my_each_with_index
+    p enum_for(:my_each_with_index) unless block_given?
+    index = 0
+    for value in self do
+      yield(value, index)
+      index += 1
+    end
+    p self
+  end
+
+  def my_select
+    p enum_for(:my_select) unless block_given?
+    new_array = []
+    for value in self do
+      new_array.push(value) if yield value
+    end
+    p new_array
+  end
+
+  def my_all?
+    conclusion = true
+    for value in self do
+      if !block_given? && !value
+        conclusion = false
+      elsif block_given? && !yield(value)
+        conclusion = false
+      elsif (value.instance_of?(Class) || value.instance_of?(Module)) && !yield(value)
+        conclusion = false
+      elsif value.instance_of?(Regexp) && !yield(value)
+      end
+  end
+    p conclusion
+  end
+
+  def my_any?
+    conclusion = false
+    for value in self do
+      if !block_given? && value
+        conclusion = true
+      elsif block_given? && yield(value)
+        conclusion = true
+      elsif (value.instance_of?(Class) || value.instance_of?(Module)) && yield(value)
+        conclusion = true
+      elsif value.instance_of?(Regexp) && yield(value)
+      end
+  end
+    p conclusion
+  end
+
+  def my_none?
+    conclusion = true
+    for value in self do
+      if !block_given? && value
+        conclusion = false
+      elsif block_given? && yield(value)
+        conclusion = false
+      elsif (value.instance_of?(Class) || value.instance_of?(Module)) && yield(value)
+        conclusion = false
+      elsif value.instance_of?(Regexp) && yield(value)
+      end
+  end
+    p conclusion
+  end
```

Comment:
#my_none? when a class is passed as an argument returns true if none of the collection is a member of such class
#my_none? when a class is passed as an argument returns true if none of the collection is a member of such class::Numeric
#my_none? when a class is passed as an argument returns false if any of the collection is a member of such class
#my_none? when a Regex is passed as an argument returns true if none of the collection matches the Regex
#my_none? when a Regex is passed as an argument returns false if any of the collection matches the Regex
#my_none? when a pattern other than Regex or a Class is given returns true only if none of the collection matches the pattern
#my_none? when a pattern other than Regex or a Class is given returns false only if one of the collection matches the pattern